### PR TITLE
distsql: fix a bug of ranger.Range to kv.KeyRange conversion

### LINKED ISSUE #40158: Query results differ from mysql with indexed null
## Bug Report

Result filtering seems broken when using an index containing `null`.

### 1. Minimal reproduce step

```sql
create table t1 (
    _id int PRIMARY KEY,
    c1 char,
    index (c1)
);

insert into t1 values (1, null);

select * from t1 where c1 is null and _id < 1;
```

### 2. What did you expect to see?
No rows returned. Using mysql (8.0.23) I get:
```
Empty set (0.00 sec)
```

### 3. What did you see instead
tidb shows a row that doesn't satisfy `_id < 1` from the where clause:
```
+-----+------+
| _id | c1   |
+-----+------+
|   1 | NULL |
+-----+------+
1 row in set (0.00 sec)
```

If I drop the index, I see tidb filters out the row correctly
```
mysql> select * from t1 where c1 is null and _id < 1;
+-----+------+
| _id | c1   |
+-----+------+
|   1 | NULL |
+-----+------+
1 row in set (0.00 sec)

mysql> explain  select * from t1 where c1 is null and _id < 1;
+------------------------+---------+-----------+------------------------+----------------------------------------------------------+
| id                     | estRows | task      | access object          | operator info                                            |
+------------------------+---------+-----------+------------------------+----------------------------------------------------------+
| IndexReader_6          | 33.23   | root      |                        | index:IndexRangeScan_5                                   |
| └─IndexRangeScan_5     | 33.23   | cop[tikv] | table:t1, index:c1(c1) | range:[NULL -inf,NULL 1), keep order:false, stats:pseudo |
+------------------------+---------+-----------+------------------------+----------------------------------------------------------+
2 rows in set (0.00 sec)

mysql> alter table t1 drop index c1;
Query OK, 0 rows affected (0.02 sec)

mysql> select * from t1 where c1 is null and _id < 1;
Empty set (0.00 sec)

mysql> explain select * from t1 where c1 is null and _id < 1;
+--------------------------+---------+-----------+---------------+------------------------------------------------+
| id                       | estRows | task      | access object | operator info                                  |
+--------------------------+---------+-----------+---------------+------------------------------------------------+
| TableReader_7            | 0.00    | root      |               | data:Selection_6                               |
| └─Selection_6            | 0.00    | cop[tikv] |               | isnull(test.t1.c1)                             |
|   └─TableRangeScan_5     | 1.00    | cop[tikv] | table:t1      | range:[-inf,1), keep order:false, stats:pseudo |
+--------------------------+---------+-----------+---------------+------------------------------------------------+
3 rows in set (0.01 sec)
```

### 4. What is your TiDB version?
```
mysql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.1.3
Edition: Community
Git Commit Hash: 6b02a5d8ba7dda28170ae2114b62e1e3b50dd974
Git Branch: heads/refs/tags/v6.1.3
UTC Build Time: 2022-11-26 07:36:17
GoVersion: go1.19.3
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

Also on tidbcloud
```
tidb_version(): Release Version: v6.3.0-serverless
Edition: Community
Git Commit Hash: 952d218e8066a04cf32bba13d536428e2695bd54
Git Branch: release-6.3-serverless
UTC Build Time: 2022-12-12 03:06:54
GoVersion: go1.19
Race Enabled: false
TiKV Min Version: 6.1.0
Check Table Before Drop: false
Store: tikv
```

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #40158

Problem Summary:

### What is changed and how it works?

The code has changed a lot since https://github.com/pingcap/tidb/pull/7163 (which introduce the bug), the old bug is fixed in some other ways.
And we can now get rid of the current bug by just resetting these changes.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
/retest-required

PTAL @winoros @qw4990

**Comment by tiancaiamao** [comment]
PTAL @winoros @qw4990

File: executor/distsql_test.go [hunk 1]
```diff
@@ -633,3 +633,14 @@ func TestCoprocessorPagingReqKeyRangeSorted(t *testing.T) {
 	tk.MustExec(`set @a=0x61219F79C90D3541F70E, @b=5501707547099269248, @c=0xEC43EFD30131DEA2CB8B, @d="呣丼蒢咿卻鹻铴础湜僂頃ǆ縍套衞陀碵碼幓9", @e="鹹楞睕堚尛鉌翡佾搁紟精廬姆燵藝潐楻翇慸嵊";`)
 	tk.MustExec(`execute stmt using @a,@b,@c,@d,@e;`)
 }
+
+func TestIssue40158(t *testing.T) {
```
Comment (hawkingrei) [inline executor/distsql_test.go:637]: Can we move this test into ```executor/issuetest``` or append it into existing test cases?

**Review by bb7133 — APPROVED** [review]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @hawkingrei

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**